### PR TITLE
SPIKE: Investigate implementing ChefStyle

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This PR is to capture conversation and investigate the amount of work that would be required to implement [chefstyle](https://github.com/chef/chefstyle) in InSpec to match [chef](https://github.com/chef/chef). After investigating, I realized it would only be an hour or so to: swap out `rubocop.yml`, run `chefstyle -a`, and fix the errant tests.

However, this process uncovered a few items that may cause issues. Mainly:
  - I'm unsure about how long it would take review this change (how much do we trust `rubocop -a`?).
  - There are a few AWS resource errors that may be valid and need to be investigated
  - We need to find a way to do the `chefstyle -a` commit as another user (or accept that one user will have the metric of changing ~17,000 lines of code)
    - I've attempted this...but I want to be 100% sure I don't get "credit" for modifying so many lines. I'm fine with signing the commit or having another author do this PR however.

This PR supersedes  #3731 due to changes in `rubocop.yml` prior to `chefstyle -a`. Thank you @miah for the initial work into this.

@clintoncwolfe @miah thoughts?

### CODE REVIEW

**Comment by jerryaldrichiii** [comment]
Looks like the functional tests aren't too hairy, more quote changes. I'll make it right.

File: examples/profile-aws/controls/iam_users_console_users_mfa.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
-control 'iam_users_console_users_mfa' do
-  title 'Users with console access must have MFA enabled'
+control "iam_users_console_users_mfa" do
+  title "Users with console access must have MFA enabled"
   impact 1.0
 
-  describe aws_iam_users.where { has_console_password and not has_mfa_enabled } do
-    its('entries') { should be_empty }
+  describe aws_iam_users.where { has_console_password && (not has_mfa_enabled) } do
```
Comment (miah) [inline examples/profile-aws/controls/iam_users_console_users_mfa.rb:5]: I'm curious if this works as expected. I know that `and` is used in lots of profiles and wonder if this is part of the rspec/inspec dsl and how this change affects that?

`and` is a [Ruby keyword](https://docs.ruby-lang.org/en/2.2.0/keywords_rdoc.html), but its also used by [rspec to create compound exceptions](https://relishapp.com/rspec/rspec-expectations/docs/compound-expectations)

**Comment by jerryaldrichiii** [comment]
Discussion from standup:
  - Need to break out failing tests from AWS
  - Need to do this change when we reduce PR count (due to merge conflicts)
  - Should we do a minor bump? (SemVer doesn't require it)

**Comment by jerryaldrichiii** [comment]
Closing this, the amount of work has been scoped and potential issues identified.

Essentially, the amount of work is relatively small, but the change touches nearly every file so we should only do this only when our outstanding PR count is low. Also, we should debate whether a minor version bump is needed when we do the work.

**Pull request closed without merging**